crawler/crawler/cloneRepository.go

```diff
@@ -33,17 +33,12 @@ func CloneRepository(domain Domain, hostname, name, gitURL, gitBranch, index str
 		if err != nil {
 			return errors.New(fmt.Sprintf("cannot git pull the repository: %s: %s", err.Error(), out))
 		}
-		// Command is: git reset --hard origin/<branch_name>
-		out, err = exec.Command("git", "-C", path, "reset", "--hard", "origin/"+gitBranch).CombinedOutput() // nolint: gas
-		if err != nil {
-			return errors.New(fmt.Sprintf("cannot git pull the repository: %s: %s", err.Error(), out))
-		}
 		return err
 	}
 
 	// Clone the repository using the external command "git".
-	// Command is: git clone -b <branch> <remote_repo>
-	out, err := exec.Command("git", "clone", "-b", gitBranch, gitURL, path).CombinedOutput() // nolint: gas
+	// Command is: git clone --filter=blob:none --mirror -b <branch> <remote_repo>
+	out, err := exec.Command("git", "clone", "--filter=blob:none", "--mirror", "-b", gitBranch, gitURL, path).CombinedOutput() // nolint: gas
```

Comment:
Does it fall back to getting the blobs if the server doesn't support the option?

Reply:
Git shows a `warning: filtering not recognized by server, ignoring` but doesn't exit with an error.